### PR TITLE
Update AddNewOP document.

### PR DESCRIPTION
**Pull request by ebarsoum**
Update the NewOP doc based on what we approved in the SIG operator meeting.

### CODE REVIEW

File: docs/AddNewOp.md [hunk 1]
```diff
@@ -1,58 +1,51 @@
...
+## Proposing a new operator
+In order to propose a new operator, the following is needed:
+1. If the operator can be composed by other ONNX operators, then it should be a function and not an operator (we have a function in ONNX : MeanVarianceNormalization).
+2. If the operators can be split to new primitives, propose those primitives instead and make the operator a function.
+3. Based on a model. This will help us understand the usage and that it solves an actual problem. For the case of the model being private or IP and can't be shared, the operator doesn't belong to the standard and should be implemented as custom OP.
+4. The operator needs to be implemented by at-least one (well-known) framework. This help us to understand the actual behavior of the operator and its usage.
+5. Operator signature and behavior:
+    1. If the operator is available in numpy, prefer numpy semantics.
+    2. If the operator is available in more than one frameworks, make sure that your design is general and cover those frameworks.
+6. Prefer attributes over inputs.
```
Comment (wschin) [inline docs/AddNewOp.md]: Is it `prefer inputs over attributes`? I am fine with both.

**Comment by gramalingam** [comment]
Looks good to me. A pragmatic question: when will we start imposing these requirements? Will it apply to updates of existing ops (e.g., fixing shape-inference-logic error in existing ops) … this might make fixes somewhat expensive. I think it might be better to fix existing ops gradually, as time permits, and not require this for bug-fixes to existing ops.

File: docs/AddNewOp.md (hunk 1 above)
Comment (ebarsoum) [reply docs/AddNewOp.md]: It is based on the SIG feedback, attributes are easy to optimize.

**Comment by ebarsoum** [comment]
@gramalingam  that is a good question, we will enforce that on current and new PR.  Let's open an issue to track them and ask community for help.

File: docs/AddNewOp.md (hunk 1 above)
Comment (wschin) [reply docs/AddNewOp.md]: Sounds good.

File: docs/AddNewOp.md [hunk 2]
```diff
@@ -1,58 +1,51 @@
...
-## Step 3: Define and polish the spec of the proposed operator.
-The spec should include:
-1. opset version of the added operator. Check out our
+# Proposing and submitting a new operator or function to ONNX
+
+Operators are the basic building blocks that define ONNX model. With a rich set of operators, ONNX can describe most DNN and ML models from various frameworks. Improving the process of accepting and adding a new operator to ONNX is crucial for ONNX standard future and its viability.
+
+In this document, we describe the process of accepting a new proposed operator and how to properly submit a new operator as part of ONNX standard. The goal is to improve on what we currently have based on our experience, learning and feedbacks we gathered from the community. 
+
+Adding new operator to ONNX is vital to keep up with latest model architectures and to keep ONNX standard relevant.
```
[the remark this answers is not in the document]
Comment (ebarsoum) [reply docs/AddNewOp.md]: You mean merge both para?

File: docs/AddNewOp.md [hunk 3]
```diff
@@ -1,58 +1,51 @@
...
-existing operators, we can define it as a Function (i.e., composite operator).
-MeanVarianceNormalization is the first example of registering an operator
-as a Function.
-
-## Step 3: Define and polish the spec of the proposed operator.
-The spec should include:
-1. opset version of the added operator. Check out our
+# Proposing and submitting a new operator or function to ONNX
+
+Operators are the basic building blocks that define ONNX model. With a rich set of operators, ONNX can describe most DNN and ML models from various frameworks. Improving the process of accepting and adding a new operator to ONNX is crucial for ONNX standard future and its viability.
```
[the remark this answers is not in the document]
Comment (ebarsoum) [reply docs/AddNewOp.md]: `proposing` isn't for the doc, it is for the operator. What the criteria for someone to propose a new operator.

File: docs/AddNewOp.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (prasanthpul) [reply docs/AddNewOp.md]: just delete this paragraph.

File: docs/AddNewOp.md [hunk 4]
```diff
@@ -2,14 +2,17 @@
 
 Operators are the basic building blocks that define ONNX model. With a rich set of operators, ONNX can describe most DNN and ML models from various frameworks. Improving the process of accepting and adding a new operator to ONNX is crucial for ONNX standard future and its viability.
 
-In this document, we describe the process of accepting a new proposed operator and how to properly submit a new operator as part of ONNX standard. The goal is to improve on what we currently have based on our experience, learning and feedbacks we gathered from the community. 
+In this document, we describe the process of accepting a new proposed operator and how to properly submit a new operator as part of ONNX standard. The goal is to improve on what we currently have based on our experience, learning and feedbacks we gathered from the community.
 
-Adding new operator to ONNX is vital to keep up with latest model architectures and to keep ONNX standard relevant.
+The ONNX specification includes a core set of operators that enable many models. It is a non-goal to add all possible operators, however more operators are added as needed to cover evolving needs.
```
Comment (prasanthpul) [inline docs/AddNewOp.md]: I suggest moving these 2 sentences to the first paragraph for better flow

File: docs/AddNewOp.md [hunk 5]
```diff
@@ -2,14 +2,17 @@
...
 
-# Accepting new operator
-The goal of ONNX standard isn't to implement every possible existing operator from all available frameworks. The goal is to cover most models and keep evolving ONNX to cover future models.
+# 4 steps to add an operator
```
Comment (prasanthpul) [inline docs/AddNewOp.md]: thanks for adding these. of course, each of these should have a section below with more details :)

Comment (ebarsoum) [reply docs/AddNewOp.md]: I thought the detail section cover the detail?

Comment (ebarsoum) [reply docs/AddNewOp.md]: Do you want to redo the entire flow?

Comment (prasanthpul) [reply docs/AddNewOp.md]: Not really. Let me submit a quick PR to your fork

File: docs/AddNewOp.md [hunk 6]
```diff
@@ -1,58 +1,58 @@
...
+
+# 4 steps to add an operator
+1. Decide what to propose
+2. Submit PR for new operator/function
+3. Review of PR by Operators SIG
+4. Merging of PR and inclusion in next ONNX release
+
+# Step 1: Proposing a new operator/function
+In order to propose a new operator/function, the following is needed:
+1. If the operator can be composed by other ONNX operators, then it should be a function and not an operator (we have a function in ONNX : MeanVarianceNormalization).
```
Comment (hobei) [inline docs/AddNewOp.md]: Based on my experience of implementing BatchNorm using this approach, the current framework lacks to full support. i.e. converting attributes to tensors etc. I think we need to make sure that we add the needed features to make this option suitable, we may even need additional operations to allow function operations to manipulate the tensors correctly

Comment (ebarsoum) [reply docs/AddNewOp.md]: @hobei can you propose what is missing?

**Pull request merged**